public/js/main_script.js

```diff
@@ -0,0 +1,108 @@
+const API_key = 'e0b8ee73-8e36-4d46-9651-177fdcb541a8';
+const API_url_pop = 'https://kinopoiskapiunofficial.tech/api/v2.2/films/top?type=TOP_100_POPULAR_FILMS&page=1';
+const API_url_search = 'https://kinopoiskapiunofficial.tech/api/v2.1/films/search-by-keyword?keyword='
+
+getMovies(API_url_pop);
+
+/**
+ * 
+ * @param {string} url 
+ */
+async function getMovies(url) {
+    const resp = await fetch(url, {
+        headers: {
+            "Content-Type": "application/json",
+            "X-API-KEY": API_key
+        }
+    });
+    const respData = await resp.json();
+    showMovies(respData)
+}
+
+/**
+ * Выбор цвета иконки в зависимости от рейтинга
+ * 
+ * @param {string} rate 
+ * @returns {string}
+ */
+function getColorByRating(rate) {
+    if (rate >= 8) {
+        return 'green_accent'
+    } else if (rate >= 6) {
+        return 'light_green'
+    } else if (rate >= 4) {
+        return 'orange'
+    } else if (rate >= 1) {
+        return 'red'
+    } else {
+        return 'grey'
+    }
+}
+
+/**
+ * Если указан рейтинг ожидания, возвращает 'Soon'
+ * Если рейтинг в принципе отсутсвует, возвращает 'N/R'
+ * 
+ * @param {string} rate 
+ * @returns {string}
+ */
+function getRating(rate) {
+    if (/^[\d\.]+%$/.test(rate)) {
+        return 'Soon'
+    } else if (rate == 'null') {
+        return 'N/R'
+    }
+    else {
+        return rate
+    }
+}
+
+/**
+ * 
+ * @param {object} data 
+ */
+function showMovies(data) {
+    const moviesEl = document.querySelector('.container');
+
+    // Высвобождение места для фильмов по запросу
+    document.querySelector('.container').innerHTML = '';
+
+    data.films.forEach(movie => {
+        const movieEl = document.createElement('div');
+        movieEl.classList.add('movie');
+        let location = "/movie_page.html?id=" + movie.filmId;
```

Comment:
Переменная никак не перезаписывается.
Она должна быть const
```suggestion
        const location = "/movie_page.html?id=" + movie.filmId;
```